From f818bfc084bbed6450c1e8bc9204cb5568b5ca43 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 20 Oct 2023 17:01:59 +0200
Subject: [PATCH 0419/1695] plat-stm32mp2: allow up to 8GB of external RAM

Default enable CFG_CORE_LARGE_PHYS_ADDR and set CFG_CORE_ARM64_PA_BITS
to 34 to allow external DDR sizes up to 8GB. This change does not permit
OP-TEE to execute above 32bit virtual addresses but allows OP-TEE
to accept and map shared memories of physical addresses above 4GByte.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2ee355c7c52d89fe0476465e1d96193cb199829a plat-stm32mp2: allow up to 8GB of external RAM]
Change-Id: I7a3965b43d7e0fbfd041c99c502e8dce847d22ac
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index c9f8ba692..8f055b04f 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -57,6 +57,10 @@ $(call force,CFG_WITH_LPAE,y)
 CFG_TZDRAM_START ?= 0x82000000
 CFG_TZDRAM_SIZE  ?= 0x02000000
 
+# Support DDR ranges up to 8GBytes (address range: 0x80000000 + DDR size)
+CFG_CORE_LARGE_PHYS_ADDR ?= y
+CFG_CORE_ARM64_PA_BITS ?= 34
+
 CFG_CORE_HEAP_SIZE ?= 262144
 CFG_CORE_RESERVED_SHM ?= n
 CFG_DTB_MAX_SIZE ?= 262144
-- 
2.47.2

